dev/com.ibm.ws.javaee.dd/src/com/ibm/ws/javaee/dd/web/WebApp.java

```diff
@@ -51,4 +62,20 @@ public interface WebApp extends ModuleDeploymentDescriptor, DeploymentDescriptor
     String getModuleName();
 
     AbsoluteOrdering getAbsoluteOrdering();
+
+    /**
+     * Verify if the web-app version supports metadata-complete elements.
+     * metadata-complete was added in version 2.5.
+     *
+     * @return true if metadata-complete is supported,
+     *         false if metadata-complete is not supported or the schema version is unknown
+     */
+    default boolean isMetadataCompleteSupported() {
+        try {
+            int intVersion = PlatformVersion.getVersionInt(getVersion());
+            return METADATA_COMPLETE_SUPPORT.contains(intVersion);
+        } catch (IllegalStateException e) {
```

Comment:
PlatformVersion.getVersionInt throws IllegalArgumentException, not IllegalStateException
```suggestion
        } catch (IllegalArgumentException e) {
```

Reply:
Ooops... autocorrect